common/system_power.c

```diff
@@ -22,3 +28,32 @@ system_reset(void)
 	}
 	panic("Failed to reset system");
 }
+
+void
+system_shutdown(void)
+{
+	struct device *pmic, *regl_axp803, *regl_sy8106a;
+
+	prcm_shutdown();
+
+	/* Disable sy8106a voltage regulator if defined. */
+	if ((regl_sy8106a = dm_get_dev_by_name("sy8106a")))
```

Comment:
You don't need to hardcode any names. Just iterate through all of the regulator subdevices using the two helpers (by index 0, and next)

Reply:
Done.